modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/IngestCommonPlugin.java

```diff
@@ -45,6 +49,10 @@
 public class IngestCommonPlugin extends Plugin implements ActionPlugin, IngestPlugin {
 
     static final Map<String, String> GROK_PATTERNS = Grok.getBuiltinPatterns();
+    static final Setting<TimeValue> GUARD_INTERVAL =
+        Setting.timeSetting("ingest.grok.guard.interval", TimeValue.timeValueSeconds(3), Setting.Property.NodeScope);
+    static final Setting<TimeValue> GUARD_MAX_EXECUTION_TIME =
+        Setting.timeSetting("ingest.grok.guard.max_execution_time", TimeValue.timeValueSeconds(5), Setting.Property.NodeScope);
```

Comment:
Is discussing these timeout values worth doing?

I took a look at some other defaults of ours, like in the low-level rest client. It sets its timeout to 1sec by default. Should we be more aggressive here?